Issues #21, #23, #36: increased code test coverage from 61% to 82% + UI enhancements

zohalmir opened the pull request:
UI tests: increased code test coverage from 61% to 82%
-issue #21 : UI test covering the floor fragment (indoor navigation)
-issue #23 : UI test covering the explore fragment
-issue #36 : UI test covering the saved places fragment
 
+ cards and buttons were updated to look alike, with the right color scheme

zohalmir committed:
UI test for indoor navigation: diffferent floors

JeremiahTee committed:
Increased code coverage from 61% to 82%

Fixed unit tests in the MapProcessor package.
Wrote two UI tests for both My Places and Explore fragment -> included in MapFragmentTest.kt

This increased the code coverage from 61% to 82%

Loujain-Al-Nakeeb committed:
UI Changes

- explore page buttons have been change to fit the ui mockups within our new color theme
-my favorite places and calendar cards have been changed
- setting indoor nav option icons have been change for the respective icon

Loujain-Al-Nakeeb committed:
UI Changes

-changed the color of the icon my_location to fit the color scheme

RobertBeaudenon committed:
Replaced hamberger menu with arrow

RobertBeaudenon committed:
Added test for the explore place model

RobertBeaudenon committed:
Testing repository explore place

zohalmir committed:
Merge branch 'ui_enhancement' into chain_directions

JeremiahTee committed:
Merge remote-tracking branch 'origin/master' into chain_directions

Amanda-Kolopa committed:
#19 Development:

- Adjusted the escalator points for vertical transportation in MB1 and MBs2

JeremiahTee committed:
Fix explore_fragment.xml that cause jenkins build to fail

jenkins build failed because of this error:

"/var/jenkins/workspace/build soen390/app/src/main/res/layout/explore_fragment.xml:29: Error: @+id/mapFragSearchBar is not a sibling in the same RelativeLayout [NotSibling]
            android:layout_below="@+id/mapFragSearchBar"
            ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~

   Explanation for issues of type "NotSibling":
   Layout constraints in a given ConstraintLayout or RelativeLayout should
   reference other views within the same relative layout (but not itself!)"

Removed the line "android:layout_below"

zohalmir committed:
modified explore cards to look like cards from other fragments

zohalmir committed:
Merge branch 'ui_enhancement' into chain_directions

JeremiahTee committed:
removing empty lines in .xml files

for consistency

JeremiahTee committed:
Merge remote-tracking branch 'origin/chain_directions' into chain_directions

JeremiahTee committed:
Fix UI, unit tests, refactoring

Refactoring based on post-build comments left by IDE
Fixed failing UI test and unit tests

NicholasNagy commented on app/src/androidTest/java/com/droidhats/campuscompass/views/FloorFragmentTest.kt:
```diff
@@ -10,9 +10,8 @@ import androidx.test.espresso.action.Press
-import androidx.test.espresso.matcher.ViewMatchers.withId
-import androidx.test.espresso.matcher.ViewMatchers.withText
+import androidx.test.espresso.matcher.ViewMatchers
+import androidx.test.espresso.matcher.ViewMatchers.*
```
We should not import the whole package. It will slow compile time.

NicholasNagy commented on app/src/test/java/com/droidhats/mapprocessor/TestProcessMap.kt:
```diff
@@ -18,15 +18,15 @@ class TestProcessMap {
 
     @Before
-    fun BeforeStuff() {
+    fun beforeStuff() {
```
let's rename this method to be more informative

zohalmir committed:
removed unnecessary toast messages

Amanda-Kolopa committed:
#20 Development:

- Updated hall8 marker for the path to lead to the stairs

NicholasNagy committed:
Enhance indoor path

- I enhanced the path drawn so that now the arrow goes on top of the icons.
- I re-added the heuristic function, because the greedy-first search was not as perfect as I had initially thought for finding the path, but I added more weight to the heuristic value, so it should still perform better in most cases.

NicholasNagy committed:
fixing code conventions

As requested in my review there were a few of our code conventions that were violated that I fixed.

The first with the * importing that I fixed, and the second was the irrelevant method name for my unit test preparation method.

NicholasNagy reviewed (approved):
I fixed the changes I left in my review, they were very minor, but everything is looking good, all tests pass and the UI looks amazing, you guys did great work!
